provider/search-provider/src/main/java/dev/ikm/tinkar/provider/search/Indexer.java

```diff
@@ -54,9 +51,9 @@ public class Indexer {
     private static final Logger LOG = LoggerFactory.getLogger(Indexer.class);
     private static final File defaultDataDirectory = new File("target/lucene/");
     private static DirectoryReader indexReader;
-    private static Directory indexDirectory;
-    private static Analyzer analyzer;
-    private static IndexWriter indexWriter;
+    static Directory indexDirectory;
```

Comment:
Could we make these three variables private and expose them with getters? That way we can prevent them from getting reassigned outside this class.

Reply:
Done in next commit